hy/compiler.py

```diff
@@ -1485,6 +1485,7 @@ def compile_with_expression(self, expr):
                                            optional_vars=thing)]
 
         ret = ctx + the_with
+        ret.contains_yield = ret.contains_yield or body.contains_yield
```

Comment:
Although Python has no `or=`, `|=` seems to do the right thing with bools, so this can be written `ret.contains_yield |= body.contains_yield`. You don't have to do that if you think it's too wacky.